### ISSUE TITLE
Semantics of ftpwho transfer_bytes

### ISSUE THREAD

**Issue by iosc82** [proftpd/proftpd#1755]
In a scenario where a file download is resumed, what are the semantics of the "transfer_bytes" field in ftpwho's JSON output?

It appears that prior to 1.3.7rc1, it used to mean "the position relative to the start of the file".
Starting with 1.3.7rc1, it appears that it is the number of bytes transferred during the current command (position relative to the restart point).

In resume scenarios, these differ.

I haven't found any mention of this in the Release Notes or NEWS of 1.3.7rc1, nor any GitHub issues addressing this.

What is the intended way to interpret this going forward? If it's the second, is there currently any provision to obtain the absolute position in the file?

**Comment by Castaglia** [comment]
The "transfer bytes" value displayed by `ftpwho` comes from the ScoreboardFile; the value is _intended_ to be similar to LogFormat's `%b` variable, _i.e._ "the number of bytes transferred for this command".  (I'll check into the changes in the 1.3.7 release cycle to see what might have changed there; my current suspicion is that it might related to the implementation of the FTP `RANG` command.)

For resumed transfers, if you need to get file offset at which the transfer was resumed, you can use something like:
```
# Include this note in the default LogFormat for handling resumed transfers
LogFormat "%h %l %u %t \"%r\" %s %b %{note:mod_xfer.file-offset}" enhanced
ExtendedLog /path/to/enhanced-xfer.log ALL enhanced
```

**Comment by iosc82** [comment]
Thank you @Castaglia for your clarification.

The log file approach you're describing is useful after a transfer completed.

For ongoing transfers, I'm assuming the ScoreboardFile (and hence, `ftpwho`) does not store the file offset?

**Comment by Castaglia** [comment]
> For ongoing transfers, I'm assuming the ScoreboardFile (and hence, `ftpwho`) does not store the file offset?

Correct.  The Scoreboard file stores the number of bytes transferred, and not the file offset (which can be different due to resumed uploads/downloads via the `REST` command).
